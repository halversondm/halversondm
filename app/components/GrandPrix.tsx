/**
 * Created by Daniel on 6/26/2016.
 */
import * as React from 'react'
import { PhotoGallery } from './PhotoGallery'
import { type ReactNode } from 'react'

export class GrandPrix extends React.Component<unknown, unknown> {
  render (): ReactNode {
    return (
            <div>
                <h2 className="text-primary">2001 Pontiac Grand Prix GTP Coupe</h2>
                <h3>Galaxy Silver Metallic / Granite Leather</h3>
                <ul>
                    <li>3.8L V6 Supercharged</li>
                    <li>66,500 miles</li>
                    <li>Purchased 9/29/2000 from Randy Reed Buick Pontiac GMC in Kansas
                        City, Missouri
                    </li>
                    <li>Power Windows</li>
                    <li>Power Door Locks</li>
                    <li>Power Steering</li>
                    <li>Power Side Mirrors</li>
                    <li>Power Drivers Seat with Lumbar Support</li>
                    <li>Power Sunroof</li>
                    <li>Heated Drivers Seat</li>
                    <li>Compass and Thermometer in Auto Dimming Rear View Mirror</li>
                    <li>Heads-Up Display</li>
                    <li>5 Windows Tinted to 35%</li>
                    <li>OnStar</li>
                    <li>16&quot; High Polished Torque-Star Wheels</li>
                    <li>Dunlop Sport A2 tires (about 20,000 miles on them)</li>
                    <li>KYB Struts with higher rate GM springs, Rear springs are from an
                        Oldsmobile Intrigue. No rear end sag!
                    </li>
                    <li>Motorsports Performance Cold Air Intake</li>
                    <li>PRJ spark plug wires</li>
                    <li>NGK TR55 plugs (replaced at 64,000 miles)</li>
                    <li>3.4‚Äù Supercharger Pulley</li>
                    <li>Intense Performance-coded PCM</li>
                    <li>180-degree thermostat</li>
                    <li>Custom cat-back exhaust with Flowmaster 40 series mufflers and
                        chrome round tips
                    </li>
                    <li>Polyurethane front motor mounts</li>
                    <li>Polyurethane front sway bar links</li>
                    <li>Buick Front Strut tower brace</li>
                    <li>Buick Rear Strut tower brace</li>
                    <li>Avital Remote Start</li>
                    <li>Factory Bose sound</li>
                    <li>Matching front, rear and side red badges by Reflective Concepts
                    </li>
                    <li>Disabled Daytime Running Lights</li>
                    <li>Optima Yellow Top Battery</li>
                    <li>Mobil 1 oil and transmission fluid</li>
                    <li>Non-Smoker car.&nbsp; Driven but not abused. NEVER been to the
                        track.
                    </li>
                </ul>
                <PhotoGallery perPage={9} totalPhotos={38} filePrefix="images/grandPrixPhotos/photo" fileSuffix=".jpg"/>
            </div>
    )
  }
}
