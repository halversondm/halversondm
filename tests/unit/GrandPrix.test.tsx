import React from "react";
import { render } from '@testing-library/react';
import GrandPrix from "../../app/components/GrandPrix";

test("GrandPrix component", () => {
    const unit = <GrandPrix />;
    const component = render(unit);
    const tree = component.container;
    expect(tree.toString()).toMatchSnapshot();
});