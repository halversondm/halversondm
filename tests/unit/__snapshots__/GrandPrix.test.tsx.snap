// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GrandPrix component 1`] = `"<div><div><h2 class=\\"text-primary\\">2001 Pontiac Grand Prix GTP Coupe</h2><h3>Galaxy Silver Metallic / Granite Leather</h3><ul><li>3.8L V6 Supercharged</li><li>66,500 miles</li><li>Purchased 9/29/2000 from Randy Reed Buick Pontiac GMC in Kansas City, Missouri</li><li>Power Windows</li><li>Power Door Locks</li><li>Power Steering</li><li>Power Side Mirrors</li><li>Power Drivers Seat with Lumbar Support</li><li>Power Sunroof</li><li>Heated Drivers Seat</li><li>Compass and Thermometer in Auto Dimming Rear View Mirror</li><li>Heads-Up Display</li><li>5 Windows Tinted to 35%</li><li>OnStar</li><li>16\\" High Polished Torque-Star Wheels</li><li>Dunlop Sport A2 tires (about 20,000 miles on them)</li><li>KYB Struts with higher rate GM springs, Rear springs are from an Oldsmobile Intrigue. No rear end sag!</li><li>Motorsports Performance Cold Air Intake</li><li>PRJ spark plug wires</li><li>NGK TR55 plugs (replaced at 64,000 miles)</li><li>3.4” Supercharger Pulley</li><li>Intense Performance-coded PCM</li><li>180-degree thermostat</li><li>Custom cat-back exhaust with Flowmaster 40 series mufflers and chrome round tips</li><li>Polyurethane front motor mounts</li><li>Polyurethane front sway bar links</li><li>Buick Front Strut tower brace</li><li>Buick Rear Strut tower brace</li><li>Avital Remote Start</li><li>Factory Bose sound</li><li>Matching front, rear and side red badges by Reflective Concepts</li><li>Disabled Daytime Running Lights</li><li>Optima Yellow Top Battery</li><li>Mobil 1 oil and transmission fluid</li><li>Non-Smoker car.  Driven but not abused. NEVER been to the track.</li></ul><div id=\\"photoGallery\\"><ul class=\\"row photo-gallery\\"><li class=\\"col-md-2 col-lg-2 col-sm-3 col-xs-4\\"><img data-i=\\"0\\" class=\\"img-fluid\\" src=\\"images/grandPrixPhotos/photo1.jpg\\"></li><li class=\\"col-md-2 col-lg-2 col-sm-3 col-xs-4\\"><img data-i=\\"1\\" class=\\"img-fluid\\" src=\\"images/grandPrixPhotos/photo2.jpg\\"></li><li class=\\"col-md-2 col-lg-2 col-sm-3 col-xs-4\\"><img data-i=\\"2\\" class=\\"img-fluid\\" src=\\"images/grandPrixPhotos/photo3.jpg\\"></li><li class=\\"col-md-2 col-lg-2 col-sm-3 col-xs-4\\"><img data-i=\\"3\\" class=\\"img-fluid\\" src=\\"images/grandPrixPhotos/photo4.jpg\\"></li><li class=\\"col-md-2 col-lg-2 col-sm-3 col-xs-4\\"><img data-i=\\"4\\" class=\\"img-fluid\\" src=\\"images/grandPrixPhotos/photo5.jpg\\"></li><li class=\\"col-md-2 col-lg-2 col-sm-3 col-xs-4\\"><img data-i=\\"5\\" class=\\"img-fluid\\" src=\\"images/grandPrixPhotos/photo6.jpg\\"></li><li class=\\"col-md-2 col-lg-2 col-sm-3 col-xs-4\\"><img data-i=\\"6\\" class=\\"img-fluid\\" src=\\"images/grandPrixPhotos/photo7.jpg\\"></li><li class=\\"col-md-2 col-lg-2 col-sm-3 col-xs-4\\"><img data-i=\\"7\\" class=\\"img-fluid\\" src=\\"images/grandPrixPhotos/photo8.jpg\\"></li><li class=\\"col-md-2 col-lg-2 col-sm-3 col-xs-4\\"><img data-i=\\"8\\" class=\\"img-fluid\\" src=\\"images/grandPrixPhotos/photo9.jpg\\"></li></ul><div class=\\"row\\"><div><button type=\\"button\\" data-id=\\"1\\" class=\\"btn btn-primary\\">Page 1</button></div><div><button type=\\"button\\" data-id=\\"2\\" class=\\"btn btn-primary\\">Page 2</button></div><div><button type=\\"button\\" data-id=\\"3\\" class=\\"btn btn-primary\\">Page 3</button></div><div><button type=\\"button\\" data-id=\\"4\\" class=\\"btn btn-primary\\">Page 4</button></div></div></div></div></div>"`;
